src/SourceBuild/content/test/Microsoft.DotNet.UnifiedBuild.Tests/NugetPackageContentTests.cs

```diff
@@ -0,0 +1,139 @@
+        var packagesArray = packages.Split(";")
+            // Nuget is unable to find fsharp or command-line-api packages for some reason
+            .Where(p => !(Path.GetFileName(Path.GetDirectoryName(p)) is "fsharp" or "command-line-api"))
+            .Select<string, object[]>(p => [p]);
+        return packagesArray;
+    }
+
+    /// <Summary>
+    /// Verifies the file layout of the source built sdk tarball to the Microsoft build.
+    /// The differences are captured in baselines/MsftToUbSdkDiff.txt.
+    /// Version numbers that appear in paths are compared but are stripped from the baseline.
+    /// This makes the baseline durable between releases.  This does mean however, entries
+    /// in the baseline may appear identical if the diff is version specific.
+    /// </Summary>
+    [Theory]
+    [MemberData(nameof(GetPackagePaths))]
+    public async Task CompareFileContents(string nugetPackagePath)
+    {
+        var fileName = Path.GetFileName(nugetPackagePath);
+        var repoName = Path.GetFileName(Path.GetDirectoryName(nugetPackagePath))!;
+        using PackageArchiveReader testPackageReader= new PackageArchiveReader(File.OpenRead(nugetPackagePath));
+        NuspecReader testNuspecReader = await testPackageReader.GetNuspecReaderAsync(CancellationToken.None);
+        var packageName = testNuspecReader.GetId();
+        var testPackageVersion = testNuspecReader.GetVersion().ToFullString();
+
+        NuGetVersion packageVersion = new NuGetVersion(testPackageVersion);
+        var packageStream = await TryDownloadPackage(packageName, packageVersion);
+        if (packageStream is null)
+            OutputHelper.LogWarningMessage($"Could not find package '{packageName}' with version '{packageVersion}'");
+        OutputHelper.WriteLine($"Found package '{packageName}' with version '{packageVersion}'");
+
+        using PackageArchiveReader packageReader = new PackageArchiveReader(packageStream);
+        NuspecReader nuspecReader = await packageReader.GetNuspecReaderAsync(CancellationToken.None);
+        ImmutableHashSet<string> baselineFiles = packageReader.GetFiles().Where(f => !_config.ExcludedFileExtensions.Contains(Path.GetExtension(f))).ToImmutableHashSet();
+        ImmutableHashSet<string> testFiles = testPackageReader.GetFiles().Where(f => !_config.ExcludedFileExtensions.Contains(Path.GetExtension(f))).ToImmutableHashSet();
+        foreach(var baseline in baselineFiles)
+        {
+            if (!testFiles.Contains(baseline))
+            {
+                OutputHelper.LogWarningMessage($"Unified build package '{packageName}' is missing file '{baseline}'");
+            }
+        }
+        foreach(var testFile in testFiles)
+        {
+            if (!baselineFiles.Contains(testFile))
+            {
+                OutputHelper.LogWarningMessage($"Unified build package '{packageName}' has additional file '{testFile}'");
```

Comment:
I'd silence these output messages as much as possible. They can create a lot of noise in the pipeline. See https://github.com/dotnet/installer/blob/main/src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSizeTest.cs#L184 for how we do it in SB.

Reply:
Updated to use `git diff` like the sdk content tests instead of logging to the console.